netbench/netbench/src/duplex.rs

```diff
@@ -125,22 +125,31 @@ impl<T: AsyncRead + AsyncWrite> super::Connection for Connection<T> {
 
             match self.inner.as_mut().poll_read(cx, &mut buf) {
                 Poll::Ready(_) => {
-                    if buf.filled().is_empty() && self.stream_opened {
```

Comment:
The main issue seems to be coming from the fact that we're checking `&& self.stream_opened` here. If this `poll_receive` function is called after it's closed, then we will never exit from the loop